allensdk/brain_observatory/filtered_running_speed/filtered_running_speed.py

```diff
@@ -0,0 +1,370 @@
+        behavior_velocities,
+        replay_velocities,
+        frame_times
+    ):
+        """
+        Concatenate all of the running speed data
+
+        Parameters
+        ----------
+        mapping_velocities: pandas.core.frame.DataFrame
+            Velocity data from mapping stimulus
+        behavior_velocities: pandas.core.frame.DataFrame
+           Velocity data from behavior stimulus
+        replay_velocities: pandas.core.frame.DataFrame
+            Velocity data from replay stimulus
+        frame_times: numpy.ndarray
+            list of the vsync times
+
+        Returns
+        -------
+        list[pd.DataFrame, pd.DataFrame]
+            concatenated velocity data, raw data
+        """
+
+        speed = np.concatenate(
+            (
+                behavior_velocities['speed'],
+                mapping_velocities['speed'],
+                replay_velocities['speed']),
+            axis=None
+        )
+
+        dx = np.concatenate(
+            (
+                behavior_velocities['dx'],
+                mapping_velocities['dx'],
+                replay_velocities['dx']),
+            axis=None
+        )
+
+        vsig = np.concatenate(
+            (
+                behavior_velocities['v_sig'],
+                mapping_velocities['v_sig'],
+                replay_velocities['v_sig']),
+            axis=None
+        )
+
+        vin = np.concatenate(
+            (
+                behavior_velocities['v_in'],
+                mapping_velocities['v_in'],
+                replay_velocities['v_in']),
+            axis=None
+        )
+
+        frame_indexes = list(
+            range(FilteredRunningSpeed.START_FRAME, len(frame_times))
+        )
+
+        velocities = pd.DataFrame(
+            {
+                "velocity": speed,
+                "net_rotation": dx,
+                "frame_indexes": frame_indexes,
+                "frame_time": frame_times
+            }
+        )
+
+        # due to an acquisition bug (the buffer of raw orientations
+        # may be updated more slowly than it is read, leading to
+        # a 0 value for the change in orientation over an interval)
+        # there may be exact zeros in the velocity.
+        velocities = velocities[~(np.isclose(velocities["net_rotation"], 0.0))]
+
+        raw_data = pd.DataFrame(
+            {"vsig": vsig, "vin": vin, "frame_time": frame_times, "dx": dx}
+        )
+
+        return velocities, raw_data
+
+    def _process_single_simulus_experiment(
+        self
+    ):
+        """
+        Process an experiment with a single simulus sessions
+        """
+
+        start_index = FilteredRunningSpeed.START_FRAME
+
+        frame_times = self._get_frame_times()
+
+        end_index = len(frame_times)
+
+        velocities, raw_data = self._extract_dx_info(
+            frame_times,
+            start_index,
+            end_index,
+            self.pkl_path
+        )
+
+        store = pd.HDFStore(self.args['output_path'])
```

Comment:
We don't typically use `ph.HDFStore` interface.
Please use 
```
with h5py.File(..., 'w') as f:
  f.create_dataset(name='running_speed', data=velocities)
  ...
```
instead

Reply:
I've fixed this, however It removed the column names. How do you add column names which is linked to the data?

Reply:
I see, that is the difference. You would need to create a dataset called `columns` and then when you read the h5 file, you would need to read the `columns` dataset when instantiating the dataframe. Maybe the current approach is better for dataframes

Reply:
I've reverted it to the existing h5 creation method in order to keep the columns. I think this is OK for now.